src/Swashbuckle.AspNetCore.Swagger/SwaggerMiddleware.cs

```diff
@@ -15,14 +17,25 @@ public class SwaggerMiddleware
         private readonly RequestDelegate _next;
         private readonly SwaggerOptions _options;
         private readonly TemplateMatcher _requestMatcher;
+        private readonly ISwaggerDocumentSerializer _swaggerDocumentSerializer;
 
         public SwaggerMiddleware(
             RequestDelegate next,
-            SwaggerOptions options)
+            SwaggerOptions options) : this (next, options, null)
+        {
+        }
+
+        public SwaggerMiddleware(
+            RequestDelegate next,
+            SwaggerOptions options,
+            IServiceProvider serviceProvider)
         {
             _next = next;
             _options = options ?? new SwaggerOptions();
             _requestMatcher = new TemplateMatcher(TemplateParser.Parse(_options.RouteTemplate), new RouteValueDictionary());
+
+            // Use IServiceProvider to retrieve the ISwaggerDocumentSerializer, because it is an optional service
+            _swaggerDocumentSerializer = serviceProvider?.GetService<ISwaggerDocumentSerializer>();
```

Comment:
I think this is self-documenting.

```suggestion
            _swaggerDocumentSerializer = serviceProvider?.GetService<ISwaggerDocumentSerializer>();
```

Reply:
I am wondering though... I have now added IServiceProvider in an overloaded constructor. I could also directly add ISwaggerDocumentSerializer. 

For the current setup, that would be fine, but not if in the future more optional interfaces are added. Then you would need overloads for each possible combination of available interfaces.

What is your preference?